### ISSUE TITLE
use settings to select which virtual things to "show" during Things page scan

### ISSUE THREAD

**Issue by kgiori** [mozilla-iot/virtual-things-adapter#56]
This add-on can be quite useful for demonstrations, and for creating simple things that control or are affected by other things using rules. But when it is enabled, I get annoyed by having to scroll through a long list of standard virtual things that I do not want to add to my gateway, whenever I decide to add a new smart home device. 

Would it be possible using the new configuration settings to allow the user to add standard capability things, similar to the ability to add new custom things? Then it would act more like the Pulse add-on. You only see things that you have configured in the add-ons settings. 

This change would have a secondary benefit of letting me create multiple on/off switches for example, whereas the current version only lets me add one of each capability type.

**Comment by mrstegeman** [comment]
This whole add-on is a bit of a hack. To me, it seems more logical to create add-ons that serve actual purposes than to continue piling more and more into this one. This was really only intended for our own development purposes initially.

Do you have ideas for simpler, one-off add-ons that would better serve your needs?

**Comment by kgiori** [comment]
The most useful dedicated "virtual thing" add-on I can imagine is an on/off switch. It could be similar conceptually to the pulse and timer add-ons, but instead just let you create on/off switches and give them each a name. Then for example I could just say "turn on the music" (virtual music switch) or click the music switch icon to play music from my smart speaker. This is  instead of opening an actual Sonos speaker thing icon and clicking the "play" boolean. Would be useful for TVs, Internet radio, or anything else with a complex thing definition where all you want to do is trigger something using a virtual on/off switch and use a rule for what happens while the switch is on.

But if you don't want to create a whole new add-on, then updating this one as noted above, where I can create as many switches as I need, would be fine.

**Comment by mrstegeman** [comment]
Several of those things already have the OnOffSwitch type, so they could already be used in that way.

I also created the Scene Control add-on for exactly this purpose.

**Comment by madb1lly** [comment]
On/Off switches is how I've used virtual things as well. The switches can be controlled by the operator or by other things, e.g. a rule to say when the heating system is on or off turns a virtual switch on or off, and that switch can also be operated directly.

I had a quick look at scenes but didn't think it really provided anything new for this, so didn't reconfigure and rewrite rules to use it. Was I mistaken?

**Comment by mrstegeman** [comment]
I totally understand that. There are two main benefits to the Scene Control add-on:
1. You're not littering your dashboard with piles of fake things, just one.
2. You don't have to build out custom OnOffSwitch devices by hand.

The main drawback is that you have one extra click to activate your scene, but really, it's one click.

**Comment by kgiori** [comment]
I like the scene controller for *automated* rule triggers or actions. It is a great addition to the add-on list. Despite that extra click burden to get to a scene, the real problem is that I don't typically open my UI to click icons to control lighting. I use voice commands. 

I have attached some screen shots of my scene controller (useful for rules/automation) and virtual switch. The scene controller doesn't help when I want to say "turn on party lighting", which is what the virtual on/off switch does well for me. I don't mind a few separate icons for switching things on/off manually. They are easier if I do use the GUI, and it's a necessity from what I can tell for voice commands, which I use daily.

![Screen Shot 2020-03-24 at 11 34 58 AM](https://user-images.githubusercontent.com/9487286/77464563-9b047680-6dc4-11ea-97c3-6f31e2792508.png)
![Screen Shot 2020-03-24 at 11 32 50 AM](https://user-images.githubusercontent.com/9487286/77464592-a35cb180-6dc4-11ea-9534-523b8672a23a.png)
![Screen Shot 2020-03-24 at 11 35 22 AM](https://user-images.githubusercontent.com/9487286/77464604-a8216580-6dc4-11ea-9587-0394ef775b86.png)

So my main request for the virtual add-on still stands -- that it allow me to pick which virtual things I want to see in my things scan list.

**Comment by mrstegeman** [comment]
Well, with 0.12, you will no longer have built-in voice control for _anything_, so that will become a feature request for voco or voice-addon. It seems like a valid request to be able to say "Activate scene X" or "Perform action Y".

**Comment by kgiori** [comment]
Yes, good point that voice add-on intent parsing could be improved for scene controller interaction.

I already unchecked auto-upgrade so that I can stick with 0.11 and the convenience of voice for now. I don't know how long it might take for the voice add-on update to DeepSpeech to be back to the level of control I already have. So I'll be patient and wait with 0.11. :)

If there's no time to tackle this enhancement request, no worries. It can sit in case any community volunteer happens upon it in the future, and finds my ideas useful to them too.
